src/main/java/org/testng/ITestNGMethod.java

```diff
@@ -154,10 +154,30 @@
 
   ITestNGMethod clone();
 
+  /**
+   * @deprecated - This method stands deprecated as of TestNG 7.0.0.
+   * Please use {@link #getRetryAnalyzer(ITestResult)} instead.
+   */
+  @Deprecated
   IRetryAnalyzer getRetryAnalyzer();
 
+  /**
+   * @deprecated - This method stands deprecated as of TestNG 7.0.0.
+   * Please use {@link #setRetryAnalyzerClass(Class)} instead.
+   */
+  @Deprecated
   void setRetryAnalyzer(IRetryAnalyzer retryAnalyzer);
 
+  default IRetryAnalyzer getRetryAnalyzer(ITestResult result) {
+    return null;

```

Comment:
`return getRetryAnalyzer()` otherwise you'll have to check null value from here and then call previous getRetryAnalyzer in the business code (where we expect to call the new method only)

Reply:
fixed